### DISCUSSION TITLE
Query collections filtered by multiple metafield values

### DISCUSSION THREAD

**Discussion by charle-connoringold** [Shopify/storefront-api-feedback#148] [Ideas & Requests]
It would be great to apply more than one metafield filter at a time.

Something like this:

      query Diamonds {
        collection(handle: "diamonds") {
          handle
          products(first: 30, filters: {productMetafields: [ {namespace: "diamond", key: "shape", value: "Round"}, {namespace: "diamond", key: "shape", value: "Square"} ] }) {
            edges {
              node {
                handle
                title
              }
            }
          }
        }
      }

**Comment by johncraigcole** [comment]
Hey @charle-connoringold 👋 you can do this today. From the [docs on combining filters](https://shopify.dev/custom-storefronts/products-collections/filter-products#step-2-combine-filters), filters of the same type are treated as an OR condition where filters of different types are treated as AND. The modification to your query shown below should allow you to get diamond products where their shape is `Round` or `Square`.

```
  query Diamonds {
    collection(handle: "diamonds") {
      handle
      products(first: 30, filters: [{productMetafield: {namespace: "diamond", key: "shape", value: "Round"}},{productMetafield: {namespace: "diamond", key: "shape", value: "Square"}}] ) {
        edges {
          node {
            handle
            title
          }
        }
      }
    }
  }
```

**Comment by charle-connoringold** [comment]
Hi @johncraigcole that appears to of done the trick!

Thanks.

**Comment by charle-connoringold** [comment]
@johncraigcole Do you also know if there is a limit to the number of filters I can apply?
